java/maven/src/org/netbeans/modules/maven/api/output/OutputUtils.java

```diff
@@ -241,8 +258,12 @@ public ProjectStacktraceOutputListener(Project project) {
         protected ClassPath getClassPath() {
             Project prj = ref.get();
             if(prj != null) {
-                ClassPath[] cp = prj.getLookup().lookup(ProjectSourcesClassPathProvider.class).getProjectClassPaths(ClassPath.EXECUTE);
-                return ClassPathSupport.createProxyClassPath(cp);
+                ProjectSourcesClassPathProvider prov = prj.getLookup().lookup(ProjectSourcesClassPathProvider.class);
+                List<ClassPath> cp = Stream.concat(
+                        Stream.of(prov.getProjectClassPaths(ClassPath.BOOT)),
+                        Stream.of(prov.getProjectClassPaths(ClassPath.EXECUTE)))
+                    .collect(Collectors.toList());
```

Comment:
it's a reallly overcomplicated (and not very efficient) way of cancatenating arrays :)
how about using something like this

static <T> T[] concat(T[] array1, T[] array2) {
    T[] result = Arrays.copyOf(array1, array1.length + array2.length);
    System.arraycopy(array2, 0, result, array1.length, array2.length);
    return result;
}

Reply:
i rather not start playing with indices unless there is a good reason for that. I am more concerned about correctness and readability here. 
A proper implementation will probably have to retain the module info somehow. But this should work in many if not most cases.

Reply:
the result has also just a handful of classpath objects.